### PR TITLE
Initialize gem

### PR DESCRIPTION
**Pull request by mkostick**
There are two commits here.  The first is the result of running ore:
`mine yeti_exception --travis --yard --version=1.0.0 --author="Yesware, Inc." --email="engineering@yesware.com"`

I intentionally did not use the Yesware ore template, since it adds internal things that won't be useful here.

The second commit includes edits to align with Yesware conventions and development ease.

### CODE REVIEW

**Comment by mkostick** [comment]
Note to reviewers:

This gem is meant to work with https://github.com/Yesware/yeti_logger, which is also a public repo, so it mirrors its use of Travis CI and the MIT license.

**Comment by mkostick** [comment]
I need to look into the CI errors.

Also, just for direction:  This gem is going to be an encapsulation of an exception pattern we started using on a new app recently.

File: lib/yeti_exception/version.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module YetiException
+  # yeti_exception version
+  VERSION = "1.0.0"
```
Comment (mike-yesware) [inline lib/yeti_exception/version.rb]: [nit] should this be lower or `pre` until you're ready for a real release version?

**Review by mike-yesware — COMMENTED** [review]
One comment

File: lib/yeti_exception/version.rb (hunk 1 above)
Comment (mkostick) [reply lib/yeti_exception/version.rb]: Sure, I can do that until I'm ready to publish

File: ChangeLog.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+### 1.0.0 / 2020-04-29
```
Comment (meghan-mcqueeney) [inline ChangeLog.md]: do you want to update this documentation to match the "pre"?

File: yeti_exception.gemspec [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      `git ls-files`.split($/).map do |subpath|
+        gem.files << File.join(submodule,subpath)
+      end
+    end
+  end
+  gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
+  gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
+  gem.require_paths = ['lib']
+
+  gem.add_development_dependency 'bundler', '~> 1.10'
```
Comment (meghan-mcqueeney) [inline yeti_exception.gemspec]: do we want restrictions like these?

**Review by meghan-mcqueeney — APPROVED** [review]

**Comment by meghan-mcqueeney** [comment]
G2G once CI passes.

**Comment by mkostick** [comment]
All review comments addressed, and CI is finally green

**Review by meghan-mcqueeney — APPROVED** [review]

**Pull request merged**